mgmt_upgrade_test.py

```diff
@@ -0,0 +1,109 @@
+import logging
+from time import sleep
+
+from sdcm.tester import ClusterTester
+from sdcm.mgmt import get_scylla_manager_tool, TaskStatus, update_config_file
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+class ManagerUpgradeTest(ClusterTester):
+    def test_upgrade(self):
+        target_upgrade_version = self.params.get('new_scylla_mgmt_repo')
+        monitor_node = self.monitors.nodes[0]
+        manager_tool = get_scylla_manager_tool(manager_node=monitor_node)
+        manager_tool.add_cluster(name="cluster_under_test", db_cluster=self.db_cluster,
+                                 auth_token=self.monitors.mgmt_auth_token)
+        current_manager_version = manager_tool.version
+
+        LOGGER.debug("Creating reoccurring backup and repair tasks")
+
+        mgr_cluster = manager_tool.get_cluster(cluster_name="cluster_under_test")
+        repair_task = mgr_cluster.create_repair_task(interval="1d")
+        repair_task_current_details = wait_until_task_finishes_return_details(repair_task)
+
+        self.update_all_agent_config_files()
+        bucket_name = self.params.get('backup_bucket_location').split()[0]
+        location_list = [f's3:{bucket_name}']
+        backup_task = mgr_cluster.create_backup_task(interval="1d", location_list=location_list,
+                                                     keyspace_list=["system_auth"])
+        backup_task_current_details = wait_until_task_finishes_return_details(backup_task)
+
+        upgrade_scylla_manager(pre_upgrade_manager_version=current_manager_version,
+                               target_upgrade_version=target_upgrade_version,
+                               manager_node=monitor_node,
+                               db_cluster=self.db_cluster)
+
+        LOGGER.debug("Checking that the previously created tasks' details have not changed")
+        manager_tool = get_scylla_manager_tool(manager_node=monitor_node)
+        # make sure that the cluster is still added to the manager
+        manager_tool.get_cluster(cluster_name="cluster_under_test")
+        validate_previous_task_details(task=repair_task, previous_task_details=repair_task_current_details)
+        validate_previous_task_details(task=backup_task, previous_task_details=backup_task_current_details)
+
+    def update_all_agent_config_files(self):
+        config_file = '/etc/scylla-manager-agent/scylla-manager-agent.yaml'
+        region_name = self.params.get('region_name').split()[0]
+        for node in self.db_cluster.nodes:
+            update_config_file(node=node, region=region_name, config_file=config_file)
+        sleep(60)
+
+
+def wait_until_task_finishes_return_details(task, wait=True, timeout=1000, step=10):
+    if wait:
+        task.wait_for_status([TaskStatus.DONE], timeout=timeout, step=step)
```

Comment:
what if the task fails? it also means that it "finished"

Reply:
fixed